themes/docsmith/static/build/js/main.js

```diff
@@ -281,23 +281,102 @@ function toggleNoteDisclosure(disclosureNote) {
+
+    $('body').scrollspy('refresh')
+}
+
+// Hugo's .TableOfContents shortcode 
+// (which relies on the Blackfriday markdown
+// renderer to compile the table of contents)
+// does not include headers from embedded 
+// shortguides. buildNav() rebuilds the nav, 
+// including embedded shortguides.
+function buildNav() {
+    function addListTagsForHeaderToNav(parentList, currentHeaderElement, previousHeaderElement) {
+        function headerLevel(headerElement) {
+            return parseInt(headerElement.prop('tagName').slice(1));
+        }
+        var currentHeaderLevel = headerLevel(currentHeaderElement);
+        var previousHeaderLevel = headerLevel(previousHeaderElement);
+    
+        function anchorElement(headerElement) {
+            return $("<a />").attr("href", "#" + headerElement.prop('id'))
+                .text($(headerElement.contents()[0]).text());
+        }
+    
+        if (currentHeaderLevel === previousHeaderLevel) {
+            var li = $("<li></li>").appendTo(parentList);
+            anchorElement(currentHeaderElement).appendTo(li);
+        }
+        else if (currentHeaderLevel > previousHeaderLevel) {
+            for (var i = 0; i < currentHeaderLevel - previousHeaderLevel; i++) {
+                parentList = $("<ul></ul>").appendTo(parentList.children("li:last"));
+                $("<li></li>").appendTo(parentList);
+            }
+            anchorElement(currentHeaderElement).appendTo(parentList.children("li:last"));
+        }
+        else if (currentHeaderLevel < previousHeaderLevel) {
+            for (var i = 0; i < previousHeaderLevel - currentHeaderLevel; i++) {
+                parentList = parentList.parent().closest("ul");
+            }
+            var li = $("<li></li>").appendTo(parentList);
+            anchorElement(currentHeaderElement).appendTo(li);
+        }
+    
+        return parentList;
+    }
+
+    var parentList = $('#TableOfContents > ul');
+    // All of .TableOfContent's elements were embedded in 
+    // #TableOfContents > ul > li
+    parentList.children('li:first').empty();
+
+    var headers = $('div#article-body')
+        .find(':header');
+    for (var i = 1; i < headers.length; i++) {
+        parentList = addListTagsForHeaderToNav(parentList, headers.eq(i), headers.eq(i-1));
+    }
+}
+
 (function($) {
 
+    buildNav();
+
     SidebarScroll = {
         init: function() {
             var tocElemID = '#doc-sidebar';
-            var toc = $(tocElemID);
-
-            var footer = $('footer');
-            var bottom = Math.round($(document).height() - footer.offset().top) + 80;
 
             /* activate scrollspy menu */
             var $body = $(document.body);
-            var navHeight = toc.outerHeight(true) + 10;
+
+            // The y position of the sticky nav element
+            var scrollSpyOffset = 20;
 
             $body.scrollspy({
                 target: tocElemID,
-                offset: navHeight
+                offset: scrollSpyOffset
```

Comment:
The scroll spy offset in the production site is actually incorrect: the old code calculated it from the height of the nav element. 

This lead to the scroll spy changing the highlighted section in the nav when the corresponding header was closer to the bottom of the page than the top (because the height of the nav is usually a large percentage of the height of the browser viewport).

https://getbootstrap.com/docs/3.3/javascript/#scrollspy-options